contracts/src/internal/HyperdriveLP.sol

```diff
@@ -564,4 +567,42 @@ abstract contract HyperdriveLP is
             _marketState.bondReserves = updatedBondReserves.toUint128();
         }
     }
+
+    function _updateLiquiditySafe(
```

Comment:
Two things:

1. This needs `NatSpec` comments. You can copy the NatSpec from the function above, and we'll need to add a `@return` field for the failure flag. You can copy the failure flag comment that we have on other `Safe` functions.
2. We shouldn't duplicate logic like this. Let's use this function implement `_updateLiquidity`. As part of that change, we may be eliminating the only logic that calls `calculateUpdateLiquidity`, and we should eliminate that.